### PR TITLE
tar callback for index parsing

### PR DESCRIPTION
**Pull request by kevinkreiser**
some suggestions on how to parse the index on the c++ side. none of this code is tested, it probably doesnt compile and its definitely not linted :smile: but the idea i think is sound and the implementation is almost completely

### CODE REVIEW

File: src/baldr/graphreader.cc [hunk 1]
```diff
@@ -36,40 +42,32 @@ tile_gone_error_t::tile_gone_error_t(std::string prefix, baldr::GraphId edgeid)
...
+    decltype(midgard::tar::contents) contents;
+    midgard::iterable_t<tile_index_entry> index(position, size);
+    for(const auto& entry : index) {
+      // informs the tar that we parsed the index, even though we write a bogus file name
+      auto inserted = contents.insert(std::to_string(entry.tile_id),
+                        std::make_pair(const_cast<char*>(file_begin + entry.offset), entry.size));
+      // store this info for ourselves so we dont have to parse tile names again below
+      tiles[entry.tile_id] = inserted.first->second;
+    }
+  };
```
Comment (kevinkreiser) [inline src/baldr/graphreader.cc:61]: i basically reverted the changes in this file and added only this bit and the updated call to the tar constructor below

File: src/baldr/graphreader.cc [hunk 2]
```diff
@@ -36,40 +42,32 @@ tile_gone_error_t::tile_gone_error_t(std::string prefix, baldr::GraphId edgeid)
...
+    
+    // get the info
+    decltype(midgard::tar::contents) contents;
+    midgard::iterable_t<tile_index_entry> index(position, size);
+    for(const auto& entry : index) {
+      // informs the tar that we parsed the index, even though we write a bogus file name
+      auto inserted = contents.insert(std::to_string(entry.tile_id),
+                        std::make_pair(const_cast<char*>(file_begin + entry.offset), entry.size));
+      // store this info for ourselves so we dont have to parse tile names again below
+      tiles[entry.tile_id] = inserted.first->second;
```
Comment (nilsnolde) [inline src/baldr/graphreader.cc:59]: sounds good @kevinkreiser thanks, much better.. `midgard::iterator_t` sounds awesome:)

I guess it only needs updating to actually return the `contents` to the tar and the right values for `tiles` (offset & size)

Comment (kevinkreiser) [reply src/baldr/graphreader.cc:59]: oh haha yeah i forgot the return haha

**Comment by kevinkreiser** [comment]
@PureW it was targetted at a branch that is a work in progress, not at master 

i always make this mistake too :smile:

**Pull request merged**